app/models/stat/anime_genre_breakdown.rb

```diff
@@ -0,0 +1,47 @@
+class Stat
+  class AnimeGenreBreakdown < Stat
+    # fully regenerate data
+    def recalculate!
+      genres = user.library_entries.eager_load(anime: :genres)
+                   .where.not(genres: { slug: nil })
+                   .group(:'genres.slug').count
+
+      self.stats_data = genres
+      stats_data['total'] = genres.values.reduce(:+)
+
+      if new_record?
+        save
+      else
+        update_attribute(:stats_data, stats_data)
+      end
```

Comment:
Wouldn't `save` work for both of these cases?

Also we might want `save!` which'll throw an error if the save fails (otherwise this might break silently)